common/app-starters-micrometer-common/src/main/java/org/springframework/cloud/stream/app/micrometer/common/SpringCloudStreamMicrometerCommonTags.java

```diff
@@ -28,21 +29,21 @@
  * instance index and guids. Later are necessary to allow discrimination and aggregation of app metrics by external
  * metrics collection and visualizaiton tools.
  *
+ * Use the management.metrics.cloud.stream.app.common.tags.enabled=false property to disable inserting those tags.
+ *
  * @author Christian Tzolov
  */
 @Configuration
+@ConditionalOnProperty(name = "management.metrics.cloud.stream.app.common.tags.enabled", havingValue = "true", matchIfMissing = true)
 public class SpringCloudStreamMicrometerCommonTags {
 
-	@Value("${spring.cloud.dataflow.cluster.name:default}")
-	private String clusterName;
-
 	@Value("${spring.cloud.dataflow.stream.name:unknown}")
 	private String streamName;
 
 	@Value("${spring.cloud.dataflow.stream.app.label:unknown}")
 	private String applicationName;
 
-	@Value("${instance.index:unknown}")
+	@Value("${instance.index:0}")
```

Comment:
Why this change?  In boot 1.5 this was part of the name of the applicaiton id and has this rather long fallback option.  In boot 2.0 there is no longer any instance.index added the the application id.

See [ContextIdApplicationContextInitializer](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/ContextIdApplicationContextInitializer.java) in Boot 2.0 and [ContextIdApplicationContextInitializer](https://github.com/spring-projects/spring-boot/blob/1.5.x/spring-boot/src/main/java/org/springframework/boot/context/ContextIdApplicationContextInitializer.java#L76) in 1.5

Reply:
Good call. Should i revert to `unknown` default or remove the `instance.index` all together? 

@jvalkeal  Aren't the deployers adding an instance index variable as well?

Reply:
we really don't have `instance.index` at all. It's either `INSTANCE_INDEX` or `spring.cloud.stream.instanceIndex` depending on a deployer. But dunno, either remove it or keep `0`. I wouldn't try to use `unknown`.

Reply:
why not follow the boot 1.5 convention
````
${vcap.application.instance_index:${spring.application.index:${server.port:${PORT:null}}}}
````
In CF deployer we set

````
.environmentVariable("SPRING_APPLICATION_INDEX", "${vcap.application.instance_index}")
````

In k8s it is
````

		String commandString = String
				.format("%s && %s", setIndexProperty("INSTANCE_INDEX"), setIndexProperty("spring.application.index"));
````
with 
````
	private String setIndexProperty(String name) {
		return String
			.format("echo %s=\"$(expr $HOSTNAME | grep -o \"[[:digit:]]*$\")\" >> /config/application" + ".properties",
				name);
	}
````
what is that exactly giving.... just a unique value, not really an index?  the dependency on hostname seems odd looking at it now.

Reply:
In local deployer it is `appInstanceEnv.put("spring.application.index", Integer.toString(i));`

Reply:
So `${spring.application.index:${INSTANCE_INDEX:null}}` maybe enough.  Not sure if upper cased `SPRING_APPLICATION_INDEX` needs to be there as well for cf.
`${vcap.application.instance_index:${spring.application.index:${INSTANCE_INDEX:null}}}` would work on CF though.